source/opt/compact_ids_pass.cpp

```diff
@@ -56,6 +56,17 @@ Pass::Status CompactIdsPass::Process() {
           }
           ++operand;
         }
+        if (inst->GetDebugScope().GetLexicalScope() != kNoDebugScope) {
+          auto it =
+              result_id_mapping.find(inst->GetDebugScope().GetLexicalScope());
+          if (it != result_id_mapping.end())
```

Comment:
In what situations will you not find the lexical scope in the map?

Reply:
I would suggestion you extract the code in the loop that from lines 37-45 into a new function "GetRemappedId" or something like that.  Then you should call that function here as well.

Reply:
done